client.go

```diff
@@ -14,6 +14,10 @@ var (
 	// ErrTimeout is an error returned when a client request does not
 	// receive a response within the client timeout duration.
 	ErrTimeout = errors.New("request timed out")
+
+	// ErrServerCrashed is an error returned when the ServerCrashedHeader header
+	// exists in the response from the server.
+	ErrServerCrashed = errors.New("server crashed upon handling request")
```

Comment:
I’m not sure I think that the client should be aware of a crash in the handler. Isn’t that better suited for a middleware that can inspect the response and return an error if needed?

Reply:
It makes sense since you can use the client without the framework server... I think I got greedy when I merged the server and client and saw them as the same :)